### PR TITLE
Cache location configurable

### LINKED ISSUE #92: Request to add rpm cache location configurable
## Problem description
Currently kiwi 8 creates a directory in /var/cache and puts downloaded rpms in there. We are wanting to contain all the files that gets created during the kiwi appliance creation process within the targetdir so that we manage them appropriately. Please provide an option to override the default behavior.

## Steps to reproduce the behaviour

verbatim from kiwi google groups:
==========================

Actually the package manager cache is to share data for all builds on 
the building host, thus I had planned to keep this as a fixed location. 
However it is possible to use another location for it, there is just 
no switch to change from the default place implemented 

If you need that please open an issue on the github issue page 

Regards, 
Marcus

### PR DESCRIPTION
**Pull request by schaefi**
Added global option --shared-cache-dir
    
The option allows to specify an alternative shared host_to_image cache directory. The default location is /var/cache/kiwi. Fixes #92

### CODE REVIEW

**Comment by schaefi** [comment]
Along with the option a cleanup of the tests were required. I'm sorry for the mixup, hope a review is still possible. Thanks

File: kiwi/defaults.py [hunk 1]
```diff
@@ -66,7 +66,12 @@ def get_shared_cache_location(self):
         The shared location is a directory which shares data from
         the image buildsystem host with the image root system.
         """
-        return 'var/cache/kiwi'
+        from .cli import Cli
+        global_args = Cli().get_global_args()
+        if global_args['--shared-cache-dir']:
+            return global_args['--shared-cache-dir']
+        else:
+            return 'var/cache/kiwi'
```
Comment (isbm) [inline kiwi/defaults.py]: Nope. Double returns are very rarely correct. The whole thing is oneliner:

```python
return Cli().get_global_args().get('--shared-cache-dir', 'var/cache/kiwi')
```

Also, if `var/cache/kiwi` is correct to be relative for some very odd reasons.

Comment (schaefi) [inline kiwi/defaults.py]: which reminds me that I need to care for this when the user specifies the location

**Comment by schaefi** [comment]
@isbm do you give us your ok wrt the latest update ? Thanks

**Comment by schaefi** [comment]
[configurable cache location](https://trello.com/c/KwBsJfBU/195-configurable-cache-location)

**Comment by schaefi** [comment]
Tested several combinations for cache-dirs and it worked nicely.

@isbm mind if I merge it ?

**Comment by schaefi** [comment]
guess it's ok to move on ;)

**Pull request merged**